server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/service/UserService.java

```diff
@@ -60,21 +61,23 @@ public void removeUser(Username username) throws UsersRepositoryException {
         usersRepository.removeUser(username);
     }
 
-    public void upsertUser(Username username, char[] password) throws UsersRepositoryException {
+    public void upsertUser(Username username, char[] password, Boolean isForced) throws UsersRepositoryException, UserConflictException {
         User user = usersRepository.getUserByName(username);
-        upsert(user, username, password);
+        upsert(user, username, password, isForced);
     }
 
     public boolean userExists(Username username) throws UsersRepositoryException {
         return usersRepository.contains(username);
     }
 
-    private void upsert(User user, Username username, char[] password) throws UsersRepositoryException {
+    private void upsert(User user, Username username, char[] password, Boolean isForced) throws UsersRepositoryException, UserConflictException {
         if (user == null) {
             usersRepository.addUser(username, new String(password));
-        } else {
+        } else if (isForced) {
             user.setPassword(new String(password));
             usersRepository.updateUser(user);
+        } else {
+            throw new UserConflictException("User " + username + " already exists.");
```

Comment:
You should be using the `UserAlreadyExistException` rather than creating a new one.